src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1419,6 +1432,17 @@ private String verifyWithWeak(String alg, Set<CryptoPrimitive> primitiveSet,
             }
             return String.format(rb.getString("with.weak"), alg);
         }
+        if (algParams != null) {
+            try {
+                LEGACY_CHECK.permits(algParams, jcp);
+                return alg;
```

Comment:
No need to return here since it will be returned on line 1445 anyway.

Reply:
Removed.